app/models/payroll_run.rb

```diff
@@ -4,6 +4,10 @@ class PayrollRun < ApplicationRecord
 
   validates :created_by, presence: true
 
+  validate :ensure_no_payroll_run_this_month, on: :create
+
+  scope :this_month, -> { where(created_at: DateTime.now.all_month) }
```

Comment:
I didn't know about `all_month`. That's cool

Reply:
All @robbiepaul! 😄